### PR TITLE
refactor any types

### PR DESCRIPTION
**Pull request by DmitryRomaniuk**

### CODE REVIEW

**Comment by apalchys** [comment]
@DmitryRomaniuk  it seems tests are failing

**Review by apalchys — APPROVED** [review]

File: server/src/routes/validators.ts [hunk 1]
```diff
@@ -29,23 +30,23 @@ export const validateGithubIdAndAccess = async (ctx: Router.RouterContext, next:
...
   const courseTaskId: string = ctx.params.courseTaskId;
   if (!courseTaskId) {
     setResponse(ctx, BAD_REQUEST, 'Incorrect [courseTaskId]');
     return;
   }
 
-  const task: any = await getCourseTask(Number(courseTaskId));
-  if (DateTime.local() > DateTime.fromJSDate(task.studentEndDate)) {
+  const task = await getCourseTask(Number(courseTaskId));
+  if (!task || DateTime.local() > DateTime.fromJSDate(new Date(task.studentEndDate))) {
```
Comment (DmitryRomaniuk) [inline server/src/routes/validators.ts:41]: +

File: server/src/rules/createUser.ts [hunk 2]
```diff
@@ -85,19 +85,19 @@ export async function createUser(profile: Profile, admin: boolean = false): Prom
...
     return result;
   })
-    .concat(taskOwner.map(t => ({ courseId: t.courseId, role: 'taskOwner' })))
+    .concat(taskOwner.map(t => ({ courseId: t.courseId, role: CourseRole.taskOwner })))
     .reduce((acc, item) => {
       if (!acc[item.courseId]) {
         acc[item.courseId] = [];
       }
-      acc[item.courseId] = union(acc[item.courseId], [item.role]) as any[];
+      acc[item.courseId]!.push(item.role);
```
Comment (DmitryRomaniuk) [inline server/src/rules/createUser.ts:100]: +

File: server/src/routes/course/score.ts [hunk 3]
```diff
@@ -100,8 +100,13 @@ export const postScore = (logger: ILogger) => async (ctx: Router.RouterContext)
...
   const session = ctx.state.user as IUserSession;
 
-  const isNotTaskOwner = !session.coursesRoles?.[courseId]?.includes('taskOwner');
-  if (mentor == null && !session.isAdmin && !session.coursesRoles?.[courseId]?.includes('manager') && isNotTaskOwner) {
+  const isNotTaskOwner = !session.coursesRoles?.[courseId]?.includes(CourseRole.taskOwner);
+  if (
+    mentor == null &&
+    !session.isAdmin &&
+    !session.coursesRoles?.[courseId]?.includes(CourseRole.manager) &&
+    isNotTaskOwner
```
Comment (DmitryRomaniuk) [inline server/src/routes/course/score.ts:108]: +

**Pull request merged**